gcsa/_services/authentication.py

```diff
@@ -107,7 +111,22 @@ def _get_credentials(
             else:
                 credentials_path = os.path.join(credentials_dir, credentials_file)
                 flow = InstalledAppFlow.from_client_secrets_file(credentials_path, scopes)
-                credentials = flow.run_local_server(host=host, port=port, bind_addr=bind_addr)
+                try:
+                    credentials = flow.run_local_server(
+                        host=host, port=port, bind_addr=bind_addr, timeout_seconds=120,
+                        open_browser=open_browser
+                    )
+                except webbrowser.Error:
+                    # System has no default browser configured, retry without opening browser
+                    return AuthenticatedService._get_credentials(
+                        token_path, credentials_dir, credentials_file, scopes, save_token, host, port,
+                        bind_addr, open_browser=False
+                    )
```

Comment:
That's a nice idea!

There are several things I'd like to change though:
1. I think `open_browser` should be an argument in the `GoogleCalendar` initialization with 3 options:
    *  `None` (default): this implementation, i.e. try with the browser. If failed, log the message (using logger, e.x. [here](https://github.com/kuzmoyev/google-calendar-simple-api/pull/192/files#diff-3a21c0a6a8138e38ef0f89b34d8a2abea9c5bea69ca8d8994e2e0107136acde8R200)) and try without the browser 
    *  `True`: try with the browser, fail fast if failed (i.e. re-raise `webbrowser.Error`)
    *  `False`: try without a browser
2. Instead of calling `_get_credentials` recursively, just use 
    ```
    credentials = flow.run_local_server(
                        host=host,
                        port=port,
                        bind_addr=bind_addr,
                        open_browser=False
                    )
    ```
    Otherwise, the logic before that line will be the same but run twice redundantly.

Reply:
1. Yes 💯 
2. The recursive call is there because of the timeout handling. If you didn't recurse, then you would either need to repeat the Timeout exception handling, or drop the timeout support commit

Reply:
Regarding 2., I see. I know nested try-except might look weird, but in this case it seems like

```python
try:
    try:
        credentials = flow.run_local_server(
            host=host, port=port, bind_addr=bind_addr, timeout_seconds=120,
            open_browser=open_browser
        )
    except webbrowser.Error:
        # System has no default browser configured, retry without opening browser
        credentials = flow.run_local_server(
            host=host, port=port, bind_addr=bind_addr, timeout_seconds=120,
            open_browser=False
        )
 except WSGITimeout:
     print('Authentication flow timed out. Please try again.')
```

would make sense.

In any case, both options would do, but that's a discussion for a future release :)